src/lem/circuit.rs

```diff
@@ -491,13 +486,70 @@ struct RecursiveContext<'a, F: LurkField, C> {
     bindings: &'a VarMap<Val>,
 }
 
+fn synthesize_call<F: LurkField, CS: ConstraintSystem<F>, C: Coprocessor<F>>(
+    cs: &mut CS,
+    func: &Func,
+    not_dummy: &Boolean,
+    next_slot: &mut SlotsCounter,
+    bound_allocations: &mut BoundAllocations<F>,
+    output_hints: &[Ptr],
+    ctx: &mut RecursiveContext<'_, F, C>,
+) -> Result<Vec<AllocatedPtr<F>>> {
+    let mut branch_outputs = BranchOutputs(vec![]);
+    synthesize_block(
+        cs,
+        &func.body,
+        &mut branch_outputs,
+        not_dummy,
+        next_slot,
+        bound_allocations,
+        ctx,
+    )?;
+    let output = allocate_return(cs, branch_outputs, ctx.store, output_hints);
+    Ok(output)
+}
+
+/// The collection of all return values and `not_dummy`s of all
+/// branches in a block.
+struct BranchOutputs<F: LurkField>(Vec<(Boolean, Vec<AllocatedPtr<F>>)>);
```

Comment:
I think this should be either `BlockOutputs` or `BranchesOutputs` (I prefer the former).

Also, why wrap it with a struct? A type alias would do